application-diagram-ui/src/main/resources/Diagram/DiagramEditSheet.xml

```diff
@@ -634,33 +634,45 @@ define('diagram-store', ['jquery', 'xwiki-meta', 'xwiki-utils', 'diagram-utils',
     }
   };
 
+  var getTagName = function(element) {
+    if (!element) { return ''; }
+    return element.tagName.toLowerCase();
+  };
+
+  var isTextElement = function(element) {
+    if (!element) { return false; }
+    return element.nodeType == 3;
+  };
+
   var isHeadingElement = function(element) {
-    var tag = element.tagName.toLowerCase();
-    return ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'].indexOf(tag) != -1;
+    if (!element) { return false; }
+    return !isTextElement(element) &amp;&amp; ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'].indexOf(getTagName(element)) != -1;
   };
 
   var isBlockElement = function(element) {
-    var tag = element.tagName.toLowerCase();
-    return ['p', 'div'].indexOf(tag) != -1 || isHeadingElement(element);
+    if (!element) { return false; }
+    return !isTextElement(element) &amp;&amp;
+      (['p', 'div'].indexOf(getTagName(element)) != -1 || isHeadingElement(element));
```

Comment:
Same, I don't think you need to check if the node is not null or not text. ``getTagName`` take care of this.